src/makeInfinityAdder.js

```diff
@@ -6,7 +6,22 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  return function sum(b) {
+    let temp;
+
+    if (b !== undefined) {
+      result += b;
+
+      return sum;
+    } else {
+      temp = result;
+      result = 0;
+
+      return temp;
+    }
```

Comment:
after the special return word, the code is no longer executed, but exits the function, so there is no need to write else
```suggestion
    if (b !== undefined) {
      result += b;

      return sum;
    }
    
    temp = result;
    result = 0;

    return temp;
```